### ISSUE TITLE
Encoder adds an extra byte on each row when image width is a multiple of 8

### ISSUE THREAD

**Issue by sesh22** [RCasatta/bmp-monochrome#2]
When the image to be encoded has a width where `width % 8 == 0`, the image becomes messed up because an extra byte is added at the end of each row of bits.

On [this line in `encode.rs`](https://github.com/RCasatta/bmp-monochrome/blob/34c04480a71f2435998667ad97a1b59d8ded82fc/src/encode.rs#L25), the number of bits to add to make a whole byte is calculated. The calculation should have an additional `% 8` to ensure that no bits get added if the width is already a whole number of bytes.
Changing this line to `writer.write(0, (8 - (width % 8) as u8) % 8)?;` seems to fix the issue.

Test code for a 8 pixel wide image:
```rust
    let data: Vec<bool> = vec![
        true, true, true, true, true, true, true, true,
        true, false, false, false, false, false, false, true,
        true, false, false, false, false, false, false, true,
        true, true, true, true, true, true, true, true,
    ];

    let bmp = bmp_monochrome::Bmp::new(data, 8)?;
    bmp.write(file)?;
```

Running this without the fix results in a single row of pixels, and with the fix results in a rectangle.

**Comment by RCasatta** [comment]
You are right! Thanks a lot for the issue, the same bug is also present in the decoder, #3 should have fixed.

I released version 0.13.0 with the fix

**Comment by sesh22** [comment]
Thanks @RCasatta, 0.13.0 fixed it for me.
I don't need to read the images so I didn't even think to check the decoder.
